zellij-server/src/ui/boundaries.rs

```diff
@@ -47,18 +48,24 @@ impl BoundarySymbol {
         self.color = color;
         *self
     }
-    pub fn as_terminal_character(&self) -> TerminalCharacter {
-        if self.invisible {
+    pub fn as_terminal_character(&self) -> Result<TerminalCharacter> {
+        let tc = if self.invisible {
             EMPTY_TERMINAL_CHARACTER
         } else {
-            let character = self.boundary_type.chars().next().unwrap();
+            let character = self.boundary_type.chars().next().with_context(|| {
+                format!(
+                    "failed to as terminal character for boundary type {}",
+                    self.boundary_type
+                )
+            })?;
```

Comment:
I think we should expand this a little. The `unwrap` here originally unwrapped on an `Option`. A `None`, however, doesn't tell us what "Error" occured here, because that is context specific. I would suggest something like this:

```rust
                .context("no boundary symbols defined")  // <-- Tells us why "None" here is actually an error for us
                .with_context(|| format!("failed to convert boundary symbol {} into terminal character", ...))?;  // <-- Gives us the usual error context
```